pydocstyle and flake8 related errors

**Describe the bug**

See pre-commit complains for pydocstyle ad flake8:

```
thoth/glyph/lib.py:44 in public function `classify_by_date`:
        D103: Missing docstring in public function
thoth/glyph/lib.py:68 in public function `classify_by_tag`:
        D103: Missing docstring in public function
thoth/glyph/lib.py:92 in public function `classify_messages`:
        D103: Missing docstring in public function
thoth/glyph/lib.py:105 in public function `classify_message`:
        D103: Missing docstring in public function
thoth/glyph/lib.py:117 in public function `generate_log`:
        D103: Missing docstring in public function
thoth/glyph/formatter.py:21 in public class `ClusterSimilar`:
        D101: Missing docstring in public class
thoth/glyph/formatter.py:24 in public method `generate_log`:
        D102: Missing docstring in public method
thoth/glyph/cli.py:99 in public function `classifybydate`:
        D103: Missing docstring in public function
thoth/glyph/cli.py:120 in public function `classifybytag`:
        D103: Missing docstring in public function
setup.py:27 in public function `get_install_requires`:
        D103: Missing docstring in public function
setup.py:33 in public function `get_version`:
        D103: Missing docstring in public function
setup.py:44 in public function `read`:
        D103: Missing docstring in public function
setup.py:64 in public method `initialize_options`:
        D102: Missing docstring in public method
setup.py:68 in public method `finalize_options`:
        D102: Missing docstring in public method
setup.py:73 in public method `run_tests`:
        D102: Missing docstring in public method
thoth/glyph/models.py:29 in public class `FasttextModel`:
        D101: Missing docstring in public class
thoth/glyph/models.py:32 in public method `classify_message`:
        D102: Missing docstring in public method
thoth/glyph/models.py:40 in public method `classify_messages`:
        D102: Missing docstring in public method
```

```
thoth/glyph/__init__.py:20:1: F401 '.lib.classify_message' imported but unused
thoth/glyph/__init__.py:21:1: F401 '.lib.classify_messages' imported but unused
thoth/glyph/__init__.py:22:1: F401 '.lib.classify_by_date' imported but unused
thoth/glyph/__init__.py:23:1: F401 '.lib.classify_by_tag' imported but unused
thoth/glyph/__init__.py:24:1: F401 '.lib.generate_log' imported but unused
thoth/glyph/__init__.py:25:1: F401 '.constants.MLModel' imported but unused
thoth/glyph/__init__.py:26:1: F401 '.constants.Format' imported but unused
thoth/glyph/__init__.py:27:1: F401 '.exceptions.RepositoryNotFoundException' imported but unused
thoth/glyph/__init__.py:28:1: F401 '.exceptions.ModelNotFoundException' imported but unused
thoth/glyph/__init__.py:29:1: F401 '.exceptions.NoMessageEnteredException' imported but unused
thoth/glyph/__init__.py:30:1: F401 '.exceptions.ThothGlyphException' imported but unused
thoth/glyph/formatter.py:26:34: E711 comparison to None should be 'if cond is not None:'
setup.py:22:1: F401 'setuptools.find_packages' imported but unused
setup.py:24:2: N812 lowercase 'test' imported as non lowercase 'TestCommand'
```

--- a/thoth/glyph/formatter.py
+++ b/thoth/glyph/formatter.py
@@ -17,14 +17,20 @@
 
 """Module containing all supported formatting options for the changelog file."""
 
+from typing import Dict
+from typing import List
+from typing import Optional
+
 
 class ClusterSimilar:
+    """Cluster similar messages into a common groups."""
 
     @staticmethod
-    def generate_log(message_dict: dict):
+    def generate_log(message_dict: Dict[str, Optional[List[str]]]) -> List[str]:
+        """Generate log out of messages stored in a dict."""
         changelog = []
         for key in message_dict:
-            if message_dict[key] != None and len(message_dict[key]) > 0:
+            if message_dict[key] is not None and len(message_dict[key]) > 0:
                 changelog.append("### " + key)
                 for message in message_dict[key]:
                     changelog.append("* " + message)

--- a/thoth/glyph/lib.py
+++ b/thoth/glyph/lib.py
@@ -43,7 +43,9 @@ DEFAULT_MODEL_PATH = os.path.join(os.path.dirname(__file__), "data/model_commits
 CHECK_PHRASES = {"Automatic Updates": ["Automatic Update of dependency"]}
 
 
-def classify_by_date(path: str, start: Optional[str] = None, end: Optional[str] = None, model: Optional[MLModel] = None) -> List[str]:
+def classify_by_date(
+    path: str, start: Optional[str] = None, end: Optional[str] = None, model: Optional[MLModel] = None
+) -> List[str]:
     """Classify commits by date."""
     start_time = 0
     end_time = sys.maxsize
@@ -68,7 +70,9 @@ def classify_by_date(path: str, start: Optional[str] = None, end: Optional[str] 
     return classify_messages(orig_messages, model)
 
 
-def classify_by_tag(path: str, start_tag: str, end_tag: Optional[str] = None, model: Optional[MLModel] = None) -> List[str]:
+def classify_by_tag(
+    path: str, start_tag: str, end_tag: Optional[str] = None, model: Optional[MLModel] = None
+) -> List[str]:
     """Classify messages for the given repo based on tags."""
     repo_path = os.path.join(path, ".git")
     if os.path.exists(repo_path):

--- a/thoth/glyph/lib.py
+++ b/thoth/glyph/lib.py
@@ -19,29 +19,32 @@
 
 import logging
 import os
-from os import path
+from typing import Dict
+from typing import List
+from typing import Optional
 
 from pygit2 import Repository
 from pygit2 import GIT_SORT_TOPOLOGICAL
 
-import time
 import datetime
 import sys
+import time
 
-from thoth.glyph import __name__
-from .exceptions import RepositoryNotFoundException
-from .exceptions import NoMessageEnteredException
-from .constants import MLModel
 from .constants import Format
-from .models import FasttextModel
+from .constants import MLModel
+from .exceptions import NoMessageEnteredException
+from .exceptions import ModelNotFoundException
+from .exceptions import RepositoryNotFoundException
 from .formatter import ClusterSimilar
+from .models import FasttextModel
 
 _LOGGER = logging.getLogger(__name__)
-DEFAULT_MODEL_PATH = path.join(path.dirname(__file__), "data/model_commits_v2_quant.bin")
+DEFAULT_MODEL_PATH = os.path.join(os.path.dirname(__file__), "data/model_commits_v2_quant.bin")
 CHECK_PHRASES = {"Automatic Updates": ["Automatic Update of dependency"]}
 
 
-def classify_by_date(path: str, start: str, end: str, model: str):
+def classify_by_date(path: str, start: Optional[str] = None, end: Optional[str] = None, model: Optional[MLModel] = None) -> List[str]:
+    """Classify commits by date."""
     start_time = 0
     end_time = sys.maxsize
 
@@ -59,13 +62,14 @@ def classify_by_date(path: str, start: str, end: str, model: str):
 
     orig_messages = []
     for commit in repo.walk(repo.head.target, GIT_SORT_TOPOLOGICAL):
-        if commit.commit_time > start_time and commit.commit_time < end_time:
+        if start_time < commit.commit_time < end_time:
             orig_messages.append(commit.message.lower())
 
     return classify_messages(orig_messages, model)
 
 
-def classify_by_tag(path: str, start_tag: str, end_tag: str, model: str):
+def classify_by_tag(path: str, start_tag: str, end_tag: Optional[str] = None, model: Optional[MLModel] = None) -> List[str]:
+    """Classify messages for the given repo based on tags."""
     repo_path = os.path.join(path, ".git")
     if os.path.exists(repo_path):
         repo = Repository(repo_path)
@@ -89,10 +93,11 @@ def classify_by_tag(path: str, start_tag: str, end_tag: str, model: str):
     return classify_messages(orig_messages, model)
 
 
-def classify_messages(messages: list, model: str):
+def classify_messages(messages: List[str], model: Optional[MLModel] = None) -> List[str]:
+    """Classify messages."""
     if messages is None or len(messages) == 0:
         _LOGGER.error("No commits found!")
-        return
+        return []
 
     if model is None:
         _LOGGER.info("Using default model")
@@ -102,7 +107,8 @@ def classify_messages(messages: list, model: str):
         return FasttextModel.classify_messages(messages)
 
 
-def classify_message(message: str, model: str) -> str:
+def classify_message(message: str, model: Optional[MLModel] = None) -> str:
+    """Classify a single message."""
     if message is None or message.strip() == "":
         raise NoMessageEnteredException
 
@@ -113,8 +119,11 @@ def classify_message(message: str, model: str) -> str:
     if model == MLModel.FASTTEXT:
         return FasttextModel.classify_message(message)
 
+    raise ModelNotFoundException(f"Unknown model: {model}")
 
-def generate_log(messages: list, format: str, model: str):
+
+def generate_log(messages: List[str], fmt: Format, model: Optional[str] = None) -> List[str]:
+    """Classify changes based on messages."""
     if not messages:
         return []
 
@@ -163,5 +172,5 @@ def generate_log(messages: list, format: str, model: str):
         _LOGGER.info("Using default format")
         model = Format.DEFAULT
 
-    if format == Format.CLUSTER_SIMILAR:
+    if fmt == Format.CLUSTER_SIMILAR:
         return ClusterSimilar.generate_log(message_dict)

--- a/thoth/glyph/models.py
+++ b/thoth/glyph/models.py
@@ -17,19 +17,23 @@
 
 """Module containing all supported Machine Learning models."""
 
+from os import path
+from typing import List
+import logging
+
 from fasttext import load_model
 import pandas as pd
-from os import path
-import logging
 
 _LOGGER = logging.getLogger(__name__)
 DEFAULT_FASTTEXT_MODEL_PATH = path.join(path.dirname(__file__), "data/model_commits_v2_quant.bin")
 
 
 class FasttextModel:
+    """A model that classifies messages using fasttext."""
 
     @staticmethod
     def classify_message(message: str) -> str:
+        """Classify a single message."""
         _LOGGER.info("Model Path : " + DEFAULT_FASTTEXT_MODEL_PATH)
         classifier = load_model(DEFAULT_FASTTEXT_MODEL_PATH)
         label = classifier.predict(message.lower())
@@ -37,7 +41,8 @@ class FasttextModel:
         return label_string
 
     @staticmethod
-    def classify_messages(messages: list):
+    def classify_messages(messages: List[str]) -> pd.DataFrame:
+        """Classify multiple messages."""
         df = pd.DataFrame(messages, columns=["message"])
         df = df.replace("\n", "", regex=True)
         commits = list(df["message"].astype(str))

